addons/udes_delivery_control/models/picking_extra_fields.py

```diff
@@ -0,0 +1,146 @@
+from odoo import api, fields, models, _
+from ..misc import date_diff, validate_dates
+
+
+class StockPickingDeliveryControl(models.Model):
+    """
+        Model to hold all extra fields of stock.picking object
+    """
+
+    _name = "udes_picking_extras"
+    _description = "UDES Picking Extra Fields"
+
+    picking_id = fields.Many2one(
+        "stock.picking", required=True, index=True, copy=False, ondelete="cascade"
+    )
+
+    # Unloading Fields
+    u_user_id = fields.Many2one("res.users", string="User", help="Picking completed by user")
+    u_delivery_note = fields.Char("Delivery Note")
+    u_location_id = fields.Many2one("stock.location", string="Location")
+    u_vehicle_scheduled_date = fields.Datetime(
+        "Scheduled Date/Time", help="Date/Time when the vehicle is expected to arrive"
+    )
+    u_vehicle_arrival_date = fields.Datetime("Arrival Date/Time", help="Arrival time of Vehicle")
+    u_week = fields.Integer("Week", compute="_compute_u_week", store=True, default=0)
+    u_unloading_start_date = fields.Datetime("Date Start", help="Unloading start date/time")
+    u_unloading_end_date = fields.Datetime("Date End", help="Unloading completed date/time")
+    u_time_taken = fields.Float(
```

Comment:
Is thin unloading time? or total?

Reply:
It is start date - end date :)

Reply:
Is that the start and end of the unloading or total? I don't know how they're collected. Eitherway we should probably rename it to reflect unloading vs total

Reply:
done